serverless/src/fetchSwodlrOrder/handler.js

```diff
@@ -0,0 +1,163 @@
+import 'array-foreach-async'
+import axios from 'axios'
+
+import { getDbConnection } from '../util/database/getDbConnection'
+import { getStateFromOrderStatus } from '../../../sharedUtils/orderStatus'
+import { parseError } from '../../../sharedUtils/parseError'
+
+const parseSwodlrResponse = (data) => {
+  const { status } = data
+  const job = status[0]
+
+  const {
+    id: jobId,
+    state,
+    reason,
+    product
+  } = job
+
+  const {
+    timestamp,
+    granules
+  } = product
+
+  return {
+    jobId,
+    status: state,
+    updatedAt: timestamp,
+    granules,
+    reason
+  }
+}
+
+const fetchSwodlrOrder = async (input) => {
+  const dbConnection = await getDbConnection()
+
+  // Destructure the payload from step function input
+  const {
+    accessToken,
+    id,
+    orderType
+  } = input
+
+  try {
+    // Fetch the retrieval id that the order belongs to so that we can provide a link to the status page
+    const retrievalOrderRecord = await dbConnection('retrieval_orders')
+      .first(
+        'retrieval_orders.order_number',
+        'retrieval_orders.order_information',
+        'retrieval_orders.state',
+        'retrieval_collections.access_method'
+      )
+      .join('retrieval_collections', { 'retrieval_orders.retrieval_collection_id': 'retrieval_collections.id' })
+      .where({
+        'retrieval_orders.id': id
+      })
+
+    // If there is not record in the database prevent taking any additional actions
+    if (!retrievalOrderRecord) {
+      return {
+        accessToken,
+        id,
+        orderStatus: 'not_found',
+        orderType
+      }
+    }
+
+    const statusGraphqlQuery = `query ($product: ID!, $limit: Int) {
```

Comment:
It's best practice to name your queries, so this opening line of the query would look like 


`query GetStatusByProduct($product: ID!, $limit: Int) {`